sdk/storage/Azure.Storage.Common/src/Shared/ClientsideEncryption/Models/EncryptionDataSerializer.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+
+namespace Azure.Storage.Cryptography.Models
+{
+    internal static class EncryptionDataSerializer
+    {
+        #region Serialize
+
+        /// <summary>
+        /// Serializes an EncryptionData instance into JSON.
+        /// </summary>
+        /// <param name="data">Data to serialize.</param>
+        /// <returns>The JSON string.</returns>
+        public static string Serialize(EncryptionData data)
+        {
+            return Encoding.UTF8.GetString(SerializeEncryptionData(data).ToArray());
+        }
+
+        /// <summary>
+        /// Serializes an EncryptionData instance into JSON.
+        /// </summary>
+        /// <param name="data">Data to serialize.</param>
+        /// <returns>The JSON UTF8 bytes.</returns>
+        public static ReadOnlyMemory<byte> SerializeEncryptionData(EncryptionData data)
```

Comment:
does this have to be public ?

Reply:
It's on an internal class.

Reply:
I mean we call it from method above. Do we call it from somewhere else? if not then we could hide it more.